tests/mcaptcha/default.nix

```diff
@@ -0,0 +1,103 @@
+{modules, ...}: let
+  port = 7000;
+  urlRoot = "http://localhost:${builtins.toString port}";
+in {
+  name = "mcaptcha tests";
```

Comment:
As far as I can tell, the two nodes `create_locally` and `bring_your_own_services` do not communicate with each other, and the VMs are completely decoupled from each other at runtime. If that's true, please split this up into two separate tests, one with one VM and one with two VMs. You may still put common configuration in a separate file an import that.

I believe you can also rather easily put shared Python code in a file and include it as a string in each of the tests.